rasterio/warp.py

```diff
@@ -241,6 +244,14 @@ def reproject(
         **kwargs)
 
 
+def res_tuple(res):
+    try:
```

Comment:
@perrygeo this is probably a style thing, but I'm not a fan of using exceptions to determine type.  What about 

```
if isinstance(res, (list, tuple)):
    return (res[0], res[0]) if len(res) == 1 else (res[0], res[1])
return ((float(res), float(res))
```

Which then seems simple enough to just inline below, and not have a separate function here.